src/linux/agent/system-server.js

```diff
@@ -85,12 +86,35 @@ rpc.exports = {
   },
   startActivity: function (pkg, activity) {
     return performOnJavaVM(function () {
-      var intent = packageManager.getLaunchIntentForPackage(pkg);
-      if (intent === null)
+      var appInstalled = false;
+      var apps = packageManager.getInstalledApplications(GET_META_DATA);
+      for (var i = 0; i < apps.size(); i++) {
```

Comment:
Let's only read the `size` once, by moving it above the loop:

```suggestion
      var numApps = apps.size();
      for (var i = 0; i !== numApps; i++) {
```

(And also use the same loop idiom as elsewhere for consistency.)